modules/andes-core/broker/src/main/java/org/wso2/andes/subscription/SubscriptionStore.java

```diff
@@ -769,6 +799,36 @@ public LocalSubscription removeLocalSubscription(LocalSubscription subscription)
     }
 
     /**
+     * Remove cluster subscriptions from database
+     *
+     * @param subscriptionToRemove The the set of andes subscriptions to be removed
+     */
+    public void removeClusterSubscriptions(Set<AndesSubscription> subscriptionToRemove) throws AndesException {
+        Iterator<AndesSubscription> iterator = subscriptionToRemove.iterator();
+        while (iterator.hasNext()) {
+            AndesSubscription subscription = iterator.next();
+            String destination = getDestination(subscription);
+            if (null != subscriptionToRemove) {
```

Comment:
If the subscriptionToRemove is null there will be a NullPointerException at line 807. Therefore no use in checking here.